Firebase/android/app/src/main/java/com/novoda/bonfire/analytics/FirebaseAnalyticsAnalytics.java

```diff
@@ -5,35 +5,100 @@
+    public void trackSignInSuccessful(String method) {
+        Bundle bundle = new Bundle();
+        bundle.putString(Param.SIGN_UP_METHOD, method);
+        firebaseAnalytics.logEvent("sign_up_success", bundle);
+    }
 
-        int length = args.length;
-        for (int i = 0; i < length; i++) {
-            Object arg = args[i];
-            builder.append(arg);
-            if (i < length - 1) {
-                builder.append(" : ");
-            }
+    @Override
+    public void trackSelectChannel(String channelName) {
+        Bundle bundle = new Bundle();
+        bundle.putString(Param.CONTENT_TYPE, "channel");
+        bundle.putString(Param.ITEM_ID, channelName);
+        firebaseAnalytics.logEvent(Event.SELECT_CONTENT, bundle);
+    }
 
-        }
-        return builder.toString();
+    @Override
+    public void trackMessageLength(int messageLength, String userId, String channelName) {
+        Bundle bundle = new Bundle();
+        bundle.putInt(Param.VALUE, messageLength);
+        bundle.putString(CHANNEL_NAME, channelName);
+        bundle.putString(USER_ID, userId);
+        firebaseAnalytics.logEvent("message_length", bundle);
     }
+
+    @Override
+    public void trackInvitationOpened(String senderId) {
+        Bundle bundle = new Bundle();
+        bundle.putString(SENDER, senderId);
+        firebaseAnalytics.logEvent("invite_opened", bundle);
+    }
+
+    @Override
+    public void trackInvitationAccepted(String senderId) {
+        Bundle bundle = new Bundle();
+        bundle.putString(SENDER, senderId);
+        firebaseAnalytics.logEvent("invite_accepted", bundle);
+    }
+
+    @Override
+    public void trackManageOwners(String userId, String channelName) {
+        Bundle bundle = new Bundle();
+        bundle.putString(USER_ID, userId);
+        bundle.putString(CHANNEL_NAME, channelName);
+        firebaseAnalytics.logEvent("manage_owners", bundle);
+    }
+
+    @Override
+    public void trackAddChannelOwner(String channelName, String userId) {
+        Bundle bundle = new Bundle();
+        bundle.putString(CHANNEL_NAME, channelName);
+        bundle.putString("added_owner", userId);
+        firebaseAnalytics.logEvent("add_channel_owner", bundle);
+    }
+
+    @Override
+    public void trackRemoveChannelOwner(String channelName, String userId) {
+        Bundle bundle = new Bundle();
+        bundle.putString(CHANNEL_NAME, channelName);
+        bundle.putString("removed_owner", userId);
+        firebaseAnalytics.logEvent("remove_channel_owner", bundle);
+    }
+
+    @Override
+    public void trackSendInvitesSelected(String userId) {
+        Bundle bundle = new Bundle();
+        bundle.putString(USER_ID, userId);
+        firebaseAnalytics.logEvent("send_invites", bundle);
+    }
+
+    @Override
+    public void trackCreateChannel(String userId) {
+        Bundle bundle = new Bundle();
+        bundle.putString(USER_ID, userId);
+        firebaseAnalytics.logEvent("create_channel", bundle);
```

Comment:
Some of the strings used in here are constants, some are just local variables in the methods themselves. Is this OK? Should we have all the strings as constants?